tests/functional/test_receipt.py

```diff
@@ -124,17 +124,28 @@ def test_get_failed_receipt(owner, vyper_contract_instance, eth_tester_provider)
 
 def test_receipt_raise_for_status_out_of_gas_error(mocker):
     gas_limit = 100000
+    txn = BaseTransaction(
```

Comment:
Shouldn't use `BaseTransaction` directly (probably should be `_BaseTransaction` to communicate that its private)

Reply:
should I import BaseTransaction as _BaseTransaction? And this is a test, we do need tests for data types that aren't supposed to be used directly, is this something that I should change, or is having this tested directly okay?

Reply:
No, it's more like we need a way to convert transaction data to the relevant subclass via `EcosystemAPI`

It might be [`create_transaction`](https://github.com/ApeWorX/ape/blob/main/src/ape/api/networks.py#L348) or it might need to be a new method

Reply:
If I'm reading this correctly, this is another ticket that has to be made?

Reply:
Yes

Reply:
[fae8941](https://github.com/ApeWorX/ape/pull/1015/commits/fae8941b98373a3c8ce0e7db9546dfa96fcdc055)

Well I solved it like this, if you don't like this, we can roll back to the commit before I did this nonsense